File: mog-rs/factory/mysql-tables-to-python-dataclass/tests/expected.py
from __future__ import annotations
from dataclasses import dataclass
from datetime import date, datetime, time
from decimal import Decimal

@dataclass
class Actor:
    actor_id: int
    first_name: str
    last_name: str
    last_update: datetime

@dataclass
class Film:
    film_id: int
    title: str
    description: str | None
    release_year: int | None
    language_id: int
    original_language_id: int | None
    rental_duration: int
    rental_rate: Decimal
    length: int | None
    replacement_cost: Decimal
    rating: str | None # TODO(mog): MySQL enum; consider a Python Enum instead of a plain str.
    special_features: str | None # TODO(mog): MySQL set; comma-joined values in one str -- consider a set[str] or list[str].
    last_update: datetime

@dataclass
class Staff:
    staff_id: int
    first_name: str
    last_name: str
    address_id: int
    picture: bytes | None
    email: str | None
    store_id: int
    active: bool
    username: str
    password: str | None
    last_update: datetime

@dataclass
class Customer:
    customer_id: int
    store_id: int
    first_name: str
    last_name: str
    email: str | None
    address_id: int
    active: bool
    create_date: datetime
    last_update: datetime | None

File: mog-rs/factory/mysql-tables-to-typescript/tests/expected.ts
export interface Actor {
  actor_id: number;
  first_name: string;
  last_name: string;
  last_update: string;
}

export interface Film {
  film_id: number;
  title: string;
  description: string | null;
  release_year: number | null;
  language_id: number;
  original_language_id: number | null;
  rental_duration: number;
  rental_rate: number;
  length: number | null;
  replacement_cost: number;
  rating: string | null; // TODO(mog): MySQL enum; consider a TS string-literal union.
  special_features: string | null; // TODO(mog): MySQL set; comma-joined values, consider string[].
  last_update: string;
}

export interface Staff {
  staff_id: number;
  first_name: string;
  last_name: string;
  address_id: number;
  picture: Uint8Array | null;
  email: string | null;
  store_id: number;
  active: boolean;
  username: string;
  password: string | null;
  last_update: string;
}

export interface Customer {
  customer_id: number;
  store_id: number;
  first_name: string;
  last_name: string;
  email: string | null;
  address_id: number;
  active: boolean;
  create_date: string;
  last_update: string | null;
}
